charts/karmada/templates/post-install-job.yaml

```diff
@@ -53,7 +53,8 @@ spec:
 
           # The `post-install hook job` may be applied before all karmada components are ready that rely on `static-resource job` and `hook-job secret`.
           # So, we have to postpone the deletion of the `static-resource job` and `hook-job secret` until all karmada components are up and running.
-          while [[ $(kubectl get pods -n {{ $namespace }} --field-selector=status.phase!=Running,status.phase!=Succeeded -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' | wc -l) > 0 ]]; do
+          while [[ $(kubectl get pods -n {{ $namespace }} --field-selector=status.phase!=Running,status.phase!=Succeeded -o jsonpath='{range .items[*]}{.metadata.name}{"\n"}{end}' | grep -v static-resource | wc -l) > 0 ]];
+          do
```

Comment:
I still don't think it's an efficient solution ( ･´ω`･ ) .

There is a misunderstanding here: the `status.phase==Running` of the karmada component means that the component has been installed successfully, actually not.

Example: `karmada-controller-manager` will soon into `running` state, but about 10 seconds lates, it finds out it miss CRD, then it will turn into `CrashLoopBackOff` state. 

I have tested a lot before, and there is a common case:

  1. `static-resource` always failed, which means it didn't successfully applied CRDs.
  2. `karmada-controller-manager` turn `running`, and several seconds later it turn into `CrashLoopBackOff` due to misssing CRD, it repeats back and forth.
  3. just during that `Running` period, the waiting condition passed and the `static-resource` job was deleted, and the CRD will no longer have chance to apply.

This is also the reason for why it used to be waiting `static-resource` job `status.phase==Succeeded`.

Reply:
I wait for the `static-resource` Job to complete before waiting for the karmada component to be ready. I can't exclude only the failed static-resource Pods in the script below. How about this?

Reply:
> I wait for the `static-resource` Job to complete before waiting for the karmada component to be ready. I can't exclude only the failed static-resource Pods in the script below.

Hold on, I just saw we have an init container for our components to wait for `static-resource` Job to be completed, So as long as the component is in the `Running` state, the `static-resource` Job must be completed, because the init container is finished, so we no longer need this, the original way is right, we can safely exclude `static-resource` Pods in post-install check script, right?

Reply:
> So as long as the component is in the Running state, the static-resource Job must be completed,

You seem to have caught the key point. I was convinced, you were right!

>  the original way is right,

right

Reply:
I have reset the commit, PTAL